src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -814,6 +818,80 @@ object Genotype {
     gt
   }
 
+  def readBiallelicDosageGenotype(isDosage: Boolean, a: ByteIterator): Double = {
+    val nAlleles = 2
+    val isBiallelic = true
+
+    val flags = a.readULEB128()
+
+    val gt: Int =
+      if (flagHasGT(isBiallelic, flags)) {
+        if (flagStoresGT(isBiallelic, flags))
+          flagGT(isBiallelic, flags)
+        else
+          a.readULEB128()
+      } else
+        -1
+
+    var count = 0
+
+    if (flagHasAD(flags)) {
+      if (flagSimpleAD(flags)) {
+        count += 1
+        val p = Genotype.gtPair(gt)
+        if (p.j != p.k)
+          count += 1
+      } else
+        count += nAlleles
+    }
+
+    if (flagHasDP(flags) && !(flagHasAD(flags) && flagSimpleDP(flags)))
+      count += 1
+
+    a.skipLEB128(count)
```

Comment:
These three blocks are exactly the same as above, right? Can we pull them out into:
```scala
def readGt(flags: Int): Int
def skipAdDp(flags: Int): Unit
```
Then this method is something like:
```scala
val gt = readGt(flags)
skipAdDp(flags)
```